homeassistant/core.py

```diff
@@ -1025,18 +1025,18 @@ def async_fire(
         listeners = self._listeners.get(event_type, [])
         match_all_listeners = self._match_all_listeners
 
+        event = Event(event_type, event_data, origin, time_fired, context)
```

Comment:
A normal setup will have a working `recorder` which subscribes to all events. Hence, we can unconditionally create the `Event` object here.